### PR TITLE
K8s 1.7 update changes. Make kubelet more customizable.  

### PR DESCRIPTION
**Pull request by puneetkatyal**
* Support custom secure port for API server
* Make kubelet options more customizable
* Use --pod-manifest-path instead of deprecated --config option
* Use new options: KUBELET_IMAGE_TAG, KUBELET_IMAGE_URL & RKT_RUN_ARGS

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -2,8 +2,17 @@
 # defaults file for ansible-coreos-kubelet
 coreos_kubelet_systemd: /etc/systemd/system
 coreos_kubelet_install_dir: /opt/bin
-kubelet_version: "v1.4.1_coreos.0"
+kubelet_version: "v1.7.0_coreos.0"
```
Comment (imkin) [inline defaults/main.yml:5]: v1.7.3_coreos.0 seems to be now available. This has a fix for https://github.com/kubernetes/kubernetes/issues/48848

File: defaults/main.yml [hunk 2]
```diff
@@ -2,8 +2,17 @@
...
 kubelet_aci: "quay.io/coreos/hyperkube"
```
Comment (imkin) [inline defaults/main.yml:6]: this should be kubelet_image_url now?

**Review by imkin — CHANGES_REQUESTED** [review]

**Comment by imkin** [comment]
Is it ok to squash the commits?

File: defaults/main.yml (hunk 2 above)
Comment (puneetkatyal) [reply defaults/main.yml:6]: No, this is only an ansible variable. We set Kubelet environment variables in `templates/kube.env.j2`

File: defaults/main.yml (hunk 1 above)
Comment (puneetkatyal) [reply defaults/main.yml:5]: That issue is for GCE. Also, I've only tested v1.7.0 so would like to leave the default to that version.
We can always update this from the base repo as needed.

**Comment by puneetkatyal** [comment]
The commits are more or less separate, so it's better to leave them as is for history.

File: defaults/main.yml (hunk 1 above)
Comment (imkin) [reply defaults/main.yml:5]: The version here does not matter actually we can override it from the coreos-ansbile vars.

**Comment by imkin** [comment]
It is difficult to get the context of the commits if they are not squashed. if they are all different, then there should have been different PR's.

**Comment by puneetkatyal** [comment]
@imkin The PR description provides enough context to the individual commits so they can be part of the same PR. I would prefer these get merged as individual commits and not as a squashed one to provide details on individual changes.

**Review by imkin — APPROVED** [review]

**Pull request merged**